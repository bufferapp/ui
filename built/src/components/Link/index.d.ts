export { default } from './Link';
//# sourceMappingURL=index.d.ts.map