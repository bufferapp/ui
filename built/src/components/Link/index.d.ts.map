{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/Link/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,QAAQ,CAAA"}