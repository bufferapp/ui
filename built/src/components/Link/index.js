export { default } from './Link';
//# sourceMappingURL=index.js.map