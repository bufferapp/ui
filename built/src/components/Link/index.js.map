{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/Link/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,QAAQ,CAAA","sourcesContent":["export { default } from './Link'\n"]}