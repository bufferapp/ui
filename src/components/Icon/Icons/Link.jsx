/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
import React from 'react';
import createIconComponent from '../utils/createIconComponent';

const LinkIcon = createIconComponent({
  content: (
    <g>
      <path d="M7.55 5.494l2.94-2.936a2.057 2.057 0 012.907 0 2.05 2.05 0 010 2.901l-2.94 2.936 1.415 1.414 2.94-2.936a4.046 4.046 0 000-5.729 4.06 4.06 0 00-5.736 0L6.136 4.08 7.55 5.494zM6.116 6.926L4.701 5.512 1.188 9.021a4.046 4.046 0 000 5.728 4.06 4.06 0 005.736 0l3.514-3.508-1.416-1.414-3.513 3.509a2.057 2.057 0 01-2.906 0 2.05 2.05 0 010-2.902l3.513-3.508z" />
      <path d="M5.787 11.732a1.015 1.015 0 01-1.434 0 1.012 1.012 0 010-1.431l5.737-5.729a1.015 1.015 0 011.434 0 1.012 1.012 0 010 1.432l-5.737 5.729z" />
    </g>
  ),
});
LinkIcon.displayName = 'LinkIcon';

export default LinkIcon;
