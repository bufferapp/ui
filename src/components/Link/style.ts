import styled from 'styled-components'
import { blue, blueDark } from '../style/colors'
import { fontFamily, fontSize, lineHeight } from '../style/fonts'

export const LinkStyled = styled.a`
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${(props) => props.fontWeight};
  line-height: ${lineHeight};
  cursor: pointer;
  text-decoration: none;
  color: ${blue};
  :hover {
    color: ${blueDark};
  }
`
