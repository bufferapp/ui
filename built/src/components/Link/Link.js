import React from 'react';
import PropTypes from 'prop-types';
import { LinkStyled } from './style';
import { fontWeightMedium } from '../style/fonts';
const Link = ({ children, href, newTab, download, className, title, fontWeight, }) => (React.createElement(LinkStyled, { href: href, target: newTab ? '_blank' : '_self', rel: newTab ? 'noopener noreferrer' : undefined, download: download, className: className, title: title, fontWeight: fontWeight }, children));
Link.propTypes = {
    children: PropTypes.node.isRequired,
    href: PropTypes.string.isRequired,
    newTab: PropTypes.bool,
    download: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    className: PropTypes.string,
    title: PropTypes.string,
    fontWeight: PropTypes.oneOf([500, 700]),
};
Link.defaultProps = {
    newTab: false,
    download: false,
    className: undefined,
    title: '',
    fontWeight: fontWeightMedium,
};
export default Link;
//# sourceMappingURL=Link.js.map