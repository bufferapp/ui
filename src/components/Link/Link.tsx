import React from 'react'
import PropTypes from 'prop-types'

import { LinkStyled } from './style'
import { fontWeightMedium } from '../style/fonts'

const Link = ({
  // @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
  children,
  // @ts-expect-error TS(7031) FIXME: Binding element 'href' implicitly has an 'any' typ... Remove this comment to see the full error message
  href,
  // @ts-expect-error TS(7031) FIXME: Binding element 'newTab' implicitly has an 'any' t... Remove this comment to see the full error message
  newTab,
  // @ts-expect-error TS(7031) FIXME: Binding element 'download' implicitly has an 'any'... Remove this comment to see the full error message
  download,
  // @ts-expect-error TS(7031) FIXME: Binding element 'className' implicitly has an 'any... Remove this comment to see the full error message
  className,
  // @ts-expect-error TS(7031) FIXME: Binding element 'title' implicitly has an 'any' ty... Remove this comment to see the full error message
  title,
  // @ts-expect-error TS(7031) FIXME: Binding element 'fontWeight' implicitly has an 'an... Remove this comment to see the full error message
  fontWeight,
}) => (
  <LinkStyled
    href={href}
    target={newTab ? '_blank' : '_self'}
    rel={newTab ? 'noopener noreferrer' : undefined}
    download={download}
    className={className}
    title={title}
    fontWeight={fontWeight}
  >
    {children}
  </LinkStyled>
)

Link.propTypes = {
  /** Linkable content. */
  children: PropTypes.node.isRequired,
  /** Url to navigate to. */
  href: PropTypes.string.isRequired,
  /** Opens new tab. */
  newTab: PropTypes.bool,
  /** Saves the linked URL instead of navigating to it. */
  download: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** Class passed by the DOM element. */
  className: PropTypes.string,
  /** Title of the link. */
  title: PropTypes.string,
  /** Font weight of the link. */
  fontWeight: PropTypes.oneOf([500, 700]),
}

Link.defaultProps = {
  newTab: false,
  download: false,
  className: undefined,
  title: '',
  fontWeight: fontWeightMedium,
}

export default Link
