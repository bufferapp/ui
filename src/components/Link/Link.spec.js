/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import Link from './Link'

snap(Link, './Link.jsx')
