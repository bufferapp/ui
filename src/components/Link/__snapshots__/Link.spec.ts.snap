// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Link Matches snapshot when boolean prop "download" is set to: "false" 1`] = `
<styled.a
  className="jest-auto-snapshots String Fixture"
  download={false}
  fontWeight="0"
  href="jest-auto-snapshots String Fixture"
  rel="noopener noreferrer"
  target="_blank"
  title="jest-auto-snapshots String Fixture"
>
  <NodeFixture />
</styled.a>
`;

exports[`jest-auto-snapshots > Link Matches snapshot when boolean prop "newTab" is set to: "false" 1`] = `
<styled.a
  className="jest-auto-snapshots String Fixture"
  download={true}
  fontWeight="0"
  href="jest-auto-snapshots String Fixture"
  target="_self"
  title="jest-auto-snapshots String Fixture"
>
  <NodeFixture />
</styled.a>
`;

exports[`jest-auto-snapshots > Link Matches snapshot when passed all props 1`] = `
<styled.a
  className="jest-auto-snapshots String Fixture"
  download={true}
  fontWeight="0"
  href="jest-auto-snapshots String Fixture"
  rel="noopener noreferrer"
  target="_blank"
  title="jest-auto-snapshots String Fixture"
>
  <NodeFixture />
</styled.a>
`;

exports[`jest-auto-snapshots > Link Matches snapshot when passed only required props 1`] = `
<styled.a
  download={false}
  fontWeight={500}
  href="jest-auto-snapshots String Fixture"
  target="_self"
  title=""
>
  <NodeFixture />
</styled.a>
`;
